muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/forum/rest/WorkspaceForumRESTService.java

```diff
@@ -473,12 +478,13 @@ public Response createThread(@PathParam ("WORKSPACEENTITYID") Long workspaceEnti
       return Response.status(Status.NOT_FOUND).entity(String.format("WorkspaceForumArea %d does not belong to workspace entity %d", forumArea.getId(), workspaceEntity.getId())).build();
     }
 
+    LockForumThread lock = newThread.getLock() != null ? LockForumThread.valueOf(newThread.getLock()) : null;
     if (sessionController.hasWorkspacePermission(ForumResourcePermissionCollection.FORUM_WRITE_WORKSPACE_MESSAGES, workspaceEntity)) {
-      if (Boolean.TRUE.equals(newThread.getSticky()) || Boolean.TRUE.equals(newThread.getLocked())) {
+      if (Boolean.TRUE.equals(newThread.getSticky()) || lock != LockForumThread.ALL) {
```

Comment:
Does lock=ALL mean (a) it's locked for all or (b) open for all? If A, shouldn't this be `lock != null` as it leads to the permission check if the user can lock or not?

Reply:
https://github.com/otavanopisto/muikku/pull/6626/files#diff-0b9d44b3ee7f314fdc158dfd4bd5a6f3951e0d87ea0ed69f4ca022f7333919f2R524

Environment forum has a different check